bugtool/cmd/helper_test.go

```diff
@@ -122,3 +122,36 @@ func (b *BugtoolSuite) TestWalkPath(c *C) {
 	err = w.walkPath(nestedDir, info, nil)
 	c.Assert(err, IsNil)
 }
+
+// TestHashEncryptionKeys tests proper hashing of keys. Lines in which `auth` or
+// other relevant pattern are found but not the hexadecimal keys are intentionally
+// removed from the output to avoid accidental leaking of keys.
+func (b *BugtoolSuite) TestHashEncryptionKeys(c *C) {
+	testStrings := []string{
+		// `auth` and hexa string
+		"<garbage> auth foo bar 0x123456af baz",
+		// `auth` but no hexa string
+		"<garbage> auth foo bar ###23456af baz",
+		// `enc` and hexa string
+		"<garbage> enc foo bar 0x123456af baz",
+		// nothing
+		"<garbage> xxxx foo bar 0x123456af baz",
+	}
+
+	validateStrings := []string{
+		// `auth` and hexa string
+		"<garbage> auth foo bar [hash:21d466b493f5c133edc008ee375e849fe5babb55d31550c25b993d151038c8a8] baz",
+		// `auth` but no hexa string
+		"",
+		// `enc` and hexa string
+		"<garbage> enc foo bar [hash:21d466b493f5c133edc008ee375e849fe5babb55d31550c25b993d151038c8a8] baz",
+		// nothing
+		"<garbage> xxxx foo bar 0x123456af baz",
+	}
+	for i, v := range testStrings {
+		modifiedString := hashEncryptionKeys(v)
+		if len(modifiedString) > 0 {
+			c.Assert(modifiedString[:len(modifiedString)-1], Equals, validateStrings[i])
```

Comment:
If we simplify the helper and just join with `\n` at the end, as suggested above, this becomes much simpler:
```suggestion
			c.Assert(modifiedString, Equals, validateStrings[i])
```

Reply:
I remember adding \n made some lint test fail. That's why I used the above method. But I'll look into it again.